### PR TITLE
Add Logger contract to allow end user implement their own Logger

### PR DESCRIPTION
**Pull request by remarkusable**
This PR enables the end user to specify their own `Logger` implementation.

There are some use cases where:
- End user want to Log into a file or cloud base logging storage
- End user want to redact some information from the Log, for example PII and GDPR related compliance. At the moment if an Exception occurred and get logged, it might contain the data in the `Message`

### CODE REVIEW

**Review by mateusjunges — CHANGES_REQUESTED** [review]
Hey 👋 Thanks for your PR. Just added a minor change request 🙂

File: src/Logger.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
 use RdKafka\Message;
 use Throwable;
 
-class Logger
+class Logger implements Contracts\Logger
```
[the remark this answers is not in the document]
Comment (remarkusable) [reply src/Logger.php:12]: do you want to make it as an Alias or rename the interface file name and move the directory?

[the remark this answers is not in the document]
Comment (remarkusable) [reply src/Logger.php:12]: looking at your PR for v2, this should be changed into:
```suggestion
class Logger implements Logger
```
and there will be an import for `\Junges\Kafka\Contracts\Logger`

**Review by mateusjunges — APPROVED** [review]
Thanks!

**Comment by mateusjunges** [comment]
I have to write docs for this so it will be released tomorrow 🙂

**Comment by mateusjunges** [comment]
Hey @remarkusable just as a FYI, released and documented [here](https://junges.dev/documentation/laravel-kafka/v1.10/advanced-usage/7-custom-loggers)

**Pull request merged**